### PR TITLE
Add livenessProbe fix to patch dashboard deployment on 1.17 upgrade

### PR DESCRIPTION
**Pull request by LaVLaS**
## Description
Adds a workaround for stalled upgrades of `deployment/rhods-dashboard` from 1.15 --> 1.17

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message.
- [x] JIRA link(s): [RHODS-5478](https://issues.redhat.com/browse/RHODS-5478)
- [ ] The Jira story is acked.
- [ ] Live build image: [quay.io/llasmith/rhods-operator-live-catalog:1.17.9-0](https://quay.io/llasmith/rhods-operator-live-catalog:1.17.9-0)
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work.
- [ ] QE contact acknowledges that this has been tested and is approved for merge.

### CODE REVIEW

**Comment by VaishnaviHire** [comment]
Code fixes look good to me. Testing the upgrade

File: deploy.sh [hunk 1]
```diff
@@ -65,6 +65,25 @@ function oc::object::safe::to::apply() {
...
+
+  # Apply 3 patch operations to run only IF spec.livenesProbe.httpGet.port == 8080 THEN delete httpGet and set tcpSocket
+  # This should prevent the operator from reconciling the deployment back to httpGet and continue with the upgrade to 1.17+
+  test_patch='{"op": "test", "path": "/spec/template/spec/containers/0/livenessProbe/httpGet/port", "value": 8080}'
+  remove_patch='{"op": "remove", "path": "/spec/template/spec/containers/0/livenessProbe/httpGet"}'
+  add_patch='{"op": "add", "path": "/spec/template/spec/containers/0/livenessProbe/tcpSocket", "value": {"port":8080}}'
+
+  patch_result=$(oc patch $object -n $ODH_PROJECT --type json -p="[${test_patch},${add_patch},${remove_patch}]" || echo "jsonPatch of $object not applied")
+
+  return $patch_result
```
Comment (VaishnaviHire) [inline deploy.sh:84]: I think this `return` needs to be updated to `echo`, given that it throws an [error](https://github.com/red-hat-data-services/odh-deployer/pull/269#issuecomment-1276061441)

Comment (LaVLaS) [reply deploy.sh]: @tarukumar @VaishnaviHire Thx. I updated the last statement to use the `echo` command

**Pull request merged**